src/core/PICA/shader_interpreter.cpp

```diff
@@ -130,15 +138,16 @@ u8 PICAShader::getIndexedSource(u32 source, u32 index) {
 }
 
 PICAShader::vec4f PICAShader::getSource(u32 source) {
-	if (source < 0x10)
+	if (source < 0x10) {
 		return inputs[source];
-	else if (source < 0x20)
+	} else if (source < 0x20) {
 		return tempRegisters[source - 0x10];
-	else if (source <= 0x7f)
-		return floatUniforms[source - 0x20];
-	else {
-		Helpers::warn("[PICA] Unimplemented source value: %X\n", source);
-		return vec4f({f24::zero(), f24::zero(), f24::zero(), f24::zero()});
+	} else {
+		const usize floatIndex = (source - 0x20) & 0x7f;
```

Comment:
Is the wrapping behavior that the & 0x7f incurs confirmed somewhere 🤔

Reply:
Confirmed here: https://www.3dbrew.org/wiki/GPU/Shader_Instruction_Set#Relative_addressing